src/com/opc_ua/opcua_local_handler.cpp

```diff
@@ -79,11 +79,13 @@ void COPC_UA_Local_Handler::run() {
   mUaServer = UA_Server_new();
   if(mUaServer) {
     UA_ServerConfig *uaServerConfig = UA_Server_getConfig(mUaServer);
-    UA_ServerConfig_setMinimal(uaServerConfig, gOpcuaServerPort, nullptr);
+    /* The original logger is needed to avoid memory leak on shutdown */
+    UA_Logger uaLogger = *uaServerConfig->logging;
```

Comment:
since you're basically overriding the logging, you should probably delete yourself the original one before assigning yours. You could do that in the `configureUAServer` method

Reply:
Yes, that was my plan in the first place. However, I encountered the following issue while doing that: In 1.4 the UA logger instance is not a global constant anymore, but it is dynamically allocated in the UA_Server_new() function. It is now referenced by a series of members of the server, e.g. config.secureChannelPKI, config.sessionPKI, etc.. I tried to delete the original logger and just point all these logger references to the Forte logger, but there is one pointer to the UA logger that I can't access from the server configuration. Freeing the memory of the UA logger then leads to a segmentation fault at some point. That is why I decided to just override the UA logger, keep a pointer to it, and then reassign the original logger at server shutdown so that the logger is properly deleted and now memory leak occurs, when terminating Forte.

Reply:
I see. then you should probably add this information as a comment, since otherwise the information gets lost and someone might try to fix it and face the issue in the future. 

is this the same issue https://github.com/open62541/open62541/issues/6769?

Reply:
I already had written a comment, but I added some more lines to give more context. Thanks!

Also, yes, this is the same issue. Applying the proposed workaround does not fix the memory leak though.